src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionMicroserviceVerticle.java

```diff
@@ -938,4 +943,47 @@ private void getThumbnails(RoutingContext event) {
             }
         });
     }
+
+    /**
+     * Downloads the {@link OriginalFile} associated with the given
+     * {@link FileAnnotation}
+     * @param event Current routing context
+     */
+    private void downloadFileAnnotation(RoutingContext event) {
+        HttpServerRequest request = event.request();
+        HttpServerResponse response = event.response();
+        final AnnotationCtx annotationCtx;
+        try {
+            annotationCtx = new AnnotationCtx(request.params(),
+                    event.get("omero.session_key"));
+        } catch (IllegalArgumentException e) {
+            if (!response.closed()) {
+                response.setStatusCode(400).end(e.getMessage());
+            }
+            return;
+        }
+        annotationCtx.injectCurrentTraceContext();
+        vertx.eventBus().<JsonObject>request(
+                ImageRegionVerticle.GET_FILE_ANNOTATION_METADATA_EVENT,
+                Json.encode(annotationCtx), new Handler<AsyncResult<Message<JsonObject>>>() {
+                    @Override
+                    public void handle(AsyncResult<Message<JsonObject>> result) {
+                        if (result.failed()) {
+                            log.error(result.cause().getMessage());
+                            response.setStatusCode(404);
+                            response.end("Could not get annotation "
+                                        + request.getParam("annotationId"));
+                            return;
+                        }
+                        JsonObject fileInfo = result.result().body();
+                        String fileName = fileInfo.getString("originalFileName");
+                        String filePath = fileInfo.getString("originalFilePath");
+                        response.headers().set("Content-Type", "application/octet-stream");
+                        response.headers().set("Content-Disposition",
+                                "attachment; filename=\"" + fileName + "\"");
+                        response.sendFile(filePath);
```

Comment:
In the scenario where `/OMERO/Files/<id>` exists but is a directory, this will leads to an exception of type

```
2025-03-11 15:27:27,496 [vert.x-eventloop-thread-8] ERROR i.v.c.h.impl.HttpServerResponseImpl - File not found: /OMERO/Files/Dir-008/8963
```

which currently results in NGINX gateway timeout.

As discussed with @knabar and @kkoz, proposing to handle this scenario more gracefully and return a [HTTP 501](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501) error code